src/components/MessageList/MessageList.tsx

```diff
@@ -563,6 +588,75 @@ const MessageListWithContext = <
     return null;
   };
 
+  const maybeCallOnStartReached = () => {
```

Comment:
I think you need to explain what the goal is here, I get what you are doing, and pretty sure we can avoid casting things to promises, but at first look this seems hacky so I am trying to understand why it's needed?

Reply:
ya looks like a very specific implementation that could be fragile

Reply:
> so I am trying to understand why it's needed

You mean why we need this `maybe` part?

Reply:
> I think you need to explain what the goal is here

Can do!!